test/injector.js

```diff
@@ -0,0 +1,88 @@
+const assert = require('assert');
+const ganache = require('ganache-cli');
+const exec = require('child_process').execSync;
+const pify = require('pify');
+const Web3 = require('web3');
+const dagPB = require('ipld-dag-pb');
+const UnixFS = require('ipfs-unixfs');
+const multihashes = require('multihashes');
+const fs = require('fs');
+
+const Simple = require('./sources/pass/simple.js');
+
+const { inject } = require('../injector');
+
+describe('injector', function(){
+  it.skip('findMetadataFile: identifies a metadata file in a group of files');
+  it.skip('findMetadataFile: errors when no metadata file is present');
+
+  it.skip('rearrangeSources: checks submitted sources against expected metadata hash');
+  it.skip('rearrangeSources: assembles submitted sources into a fileName-to-content map');
+
+  it.skip('storeData: writes metadata and sources to the repository');
+  it.skip('storeData: errors if the recompiled bytecode has no metadata hash')
+
+  describe('inject', function(){
+    let server;
+    let accounts;
+    let port = 8545;
+    let chain = 'localhost';
+    let mockRepo = 'mockRepository';
+
+    before(async function(){
+      server = ganache.server();
+      await pify(server.listen)(port);
+      web3 = new Web3(`http://${chain}:${port}`);
+      accounts = await web3.eth.getAccounts();
+    })
+
+    // Clean up repository
+    afterEach(function(){
+      try { exec(`rm -rf ${mockRepo}`) } catch(err) { /*ignore*/ }
+    })
+
+    // Clean up server
+    after(async function(){
+      await pify(server.close)();
+    });
+
+    it('verifies sources from metadata with an address & stores by IPFS hash', async function(){
+      this.timeout(15000);
+
+      const source = Simple.sourceCodes['Simple.sol'];
+      const metadata = Simple.compilerOutput.metadata;
+      const abi = Simple.compilerOutput.abi;
+
+      const options = {
+        data: Simple.compilerOutput.evm.bytecode.object,
+        gasPrice: '1',
+        gas: 4000000,
+      };
+
+      // Deploy contract
+      const simple = new web3.eth.Contract(abi, options);
+      const instance = await simple.deploy().send({from: accounts[0]});
```

Comment:
This is only needed so that the bytecode query returns the right value, is that true?
If so, might it make sense to just mock that json-rpc call?
My concern is that actually deploying a cotnract might also trigger the monitor and not just the injector.

Reply:
Yes, true. The monitor is not running in these tests. Agree it's important to isolate them though. 

I'd rather not mock unless necessary because it requires that the test know about internal implementation details of the method it's testing.

Reply:
Ok!